cylc/flow/run_modes/skip.py

```diff
@@ -0,0 +1,166 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Utilities supporting skip modes
+"""
+from logging import INFO
+from typing import (
+    TYPE_CHECKING, List, Set, Tuple)
+
+from cylc.flow import LOG
+from cylc.flow.exceptions import WorkflowConfigError
+from cylc.flow.platforms import get_platform
+from cylc.flow.task_outputs import (
+    TASK_OUTPUT_SUBMITTED,
+    TASK_OUTPUT_SUCCEEDED,
+    TASK_OUTPUT_FAILED,
+    TASK_OUTPUT_STARTED
+)
+from cylc.flow.task_state import RunMode
+
+if TYPE_CHECKING:
+    from cylc.flow.taskdef import TaskDef
+    from cylc.flow.task_job_mgr import TaskJobManager
+    from cylc.flow.task_proxy import TaskProxy
+    from typing_extensions import Literal
+
+
+def submit_task_job(
+    task_job_mgr: 'TaskJobManager',
+    itask: 'TaskProxy',
+    now: Tuple[float, str]
+) -> 'Literal[True]':
+    """Submit a task in skip mode.
+
+    Returns:
+        True - indicating that TaskJobManager need take no further action.
+    """
+    task_job_mgr._set_retry_timers(itask, itask.tdef.rtconfig)
+    itask.summary['started_time'] = now[0]
+    itask.waiting_on_job_prep = False
+    itask.submit_num += 1
+
+    itask.platform = get_platform()
+    itask.platform['name'] = RunMode.SKIP
+    itask.summary['job_runner_name'] = RunMode.SKIP
+    itask.tdef.run_mode = RunMode.SKIP
+    task_job_mgr.task_events_mgr.process_message(
+        itask, INFO, TASK_OUTPUT_SUBMITTED,
+    )
+    task_job_mgr.workflow_db_mgr.put_insert_task_jobs(
+        itask, {
+            'time_submit': now[1],
+            'try_num': itask.get_try_num(),
+            'flow_nums': str(list(itask.flow_nums)),
+            'is_manual_submit': itask.is_manual_submit,
+            'job_runner_name': RunMode.SIMULATION,
+            'platform_name': RunMode.SIMULATION,
+            'submit_status': 0   # Submission has succeeded
+        }
+    )
+    for output in process_outputs(itask):
+        task_job_mgr.task_events_mgr.process_message(itask, INFO, output)
+
+    return True
+
+
+def process_outputs(itask: 'TaskProxy') -> List[str]:
+    """Process Skip Mode Outputs:
+
+    * By default, all required outputs will be generated plus succeeded
+      if success is optional.
+    * The outputs submitted and started are always produced and do not
+      need to be defined in outputs.
+    * If outputs is specified and does not include either
+      succeeded or failed then succeeded will be produced.
+
+    Return:
+        A list of outputs to emit.
```

Comment:
Section (e.g. `Args:`, `Returns:`, `Examples:`) are followed with a blankline.

```suggestion
        A list of outputs to emit.

```